### ISSUE TITLE
Notification handler 'textDocument/didOpen' failed with message: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by jamespacileo** [angular/vscode-ng-language-service#120]
Not sure if this error is appearing since the last update, anyway I'm hitting this error.

    Angular Language Service: 4.3.0-beta.0-1fc0d05
    TypeScript: 2.3.4
    SERVER ERROR: Cannot read property 'kind' of undefined
    TypeError: Cannot read property 'kind' of undefined
        at Evaluator.evaluateNode (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45038:21)
        at Evaluator.evaluateNode (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45177:34)
        at C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45057:39
        at visitEachNode (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\typescript\lib\typescript.js:14785:30)
        at Object.forEachChild (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\typescript\lib\typescript.js:14901:24)
        at Evaluator.evaluateNode (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45042:20)
        at C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45139:103
        at Array.map (native)
        at Evaluator.evaluateNode (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45139:69)
        at _loop_1 (C:\Users\user\.vscode-insiders\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:45965:57)
    [Error - 20:17:43] Notification handler 'textDocument/didOpen' failed with message: Cannot read property 'kind' of undefined

Let me know if I can provde useful information.

**Comment by pdestailleur** [comment]
Having a simular issue.

```
Angular Language Service: 4.3.0-beta.0-1fc0d05
TypeScript: 2.3.4
SERVER ERROR: Cannot read property 'length' of undefined
TypeError: Cannot read property 'length' of undefined
    at StaticSymbolResolver.resolveSymbol (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:42004:33)
    at StaticReflector.getTypeMetadata (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:41814:67)
    at StaticReflector.annotations (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:41243:55)
    at DirectiveResolver.isDirective (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:31287:61)
    at CompileMetadataResolver.isDirective (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:32251:37)
    at CompileMetadataResolver._getTypeDescriptor (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:32480:18)
    at C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:32351:78
    at Array.forEach (native)
    at CompileMetadataResolver.getNgModuleMetadata (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:32334:49)
    at addNgModule (C:\Users\destapa\.vscode\extensions\Angular.ng-template-0.1.4\server\node_modules\@angular\language-service\bundles\language-service.umd.js:41114:58)
[Error - 13:32:46] Notification handler 'textDocument/didOpen' failed with message: Cannot read property 'length' of undefined
```

**Comment by chuckjaz** [comment]
@pdestailleur The stack is unrelated. Can you create a separate bug.

**Comment by chuckjaz** [comment]
@jamespacileo This is a bug in the metadata collector and I am working on a fix for it.

It appears if you have an expression in metadata that is of the form `a[]`. This is invalid but the language service should not crash if this is encountered.

**Comment by chuckjaz** [comment]
https://github.com/angular/angular/issues/17993

**Comment by scote** [comment]
Do you know if this is the same error? 

I got this one on my project when i'm saving a file.
```
Search path: c:/git/ng2-materialize/demo-app/src/app/validation/validation-ngmodel
Config file name: c:/git/ng2-materialize/demo-app/tsconfig.json
Angular Language Service: 4.2.2
TypeScript: 2.3.4
SERVER ERROR: Cannot read property 'length' of undefined
TypeError: Cannot read property 'length' of undefined
    at StaticSymbolResolver.resolveSymbol (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:42005:33)
    at StaticReflector.getTypeMetadata (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:41815:67)
    at StaticReflector.annotations (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:41244:55)
    at DirectiveResolver.isDirective (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:31290:61)
    at CompileMetadataResolver.isDirective (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:32254:37)
    at CompileMetadataResolver._getTypeDescriptor (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:32483:18)
    at c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:32354:78
    at Array.forEach (native)
    at CompileMetadataResolver.getNgModuleMetadata (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:32337:49)
    at addNgModule (c:\git\ng2-materialize\demo-app\node_modules\@angular\language-service\bundles\language-service.umd.js:41115:58)
[Error - 9:31:35 PM] Notification handler 'textDocument/didSave' failed with message: Cannot read property 'length' of undefined

```

**Comment by chuckjaz** [comment]
Closed with 0.1.6
